config.go

```diff
@@ -688,6 +693,26 @@ func ValidateConfig(cfg Config, usageMessage string,
 		return nil
 	}
 
+	// IsSet returns true if an option has been set in either the config
+	// file or by a flag.
+	isSet := func(field string) bool {
+		fieldname, ok := reflect.TypeOf(Config{}).FieldByName(field)
+		if !ok {
+			fmt.Fprintf(os.Stderr, "could not find field %s\n", field)
+			return false
+		}
+
+		long, ok := fieldname.Tag.Lookup("long")
+		if !ok {
+			fmt.Fprintf(os.Stderr,
+				"field %s does not have a long tag\n", field)
+			return false
+		}
+
+		return fileParser.FindOptionByLongName(long).IsSet() ||
```

Comment:
This can cause a panic if this `lnd` config is actually embedded in _another_ config. So we need something more generalized here...